.circleci/config.yml

```diff
@@ -0,0 +1,211 @@
+          name: install dependencies
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install flake8==2.5.4 flake8-quotes==0.11.0
+      - save_cache:
+          paths:
+            - ./venv
+          key: v1-lint-deps
+      - run:
+          name: lint code
+          command: |
+            . venv/bin/activate
+            flake8 --inline-quotes "double" pinax
+
+  py27dj18:
+    <<: *common
+    docker:
+      - image: circleci/python:2.7
+        environment:
+          DJANGO=Django>=1.8,<1.9
+  py27dj19:
+    <<: *common
+    docker:
+      - image: circleci/python:2.7
+        environment:
+          DJANGO=Django>=1.9,<1.10
+  py27dj110:
+    <<: *common
+    docker:
+      - image: circleci/python:2.7
+        environment:
+          DJANGO=Django>=1.10,<1.11
+  py27dj111:
+    <<: *common
+    docker:
+      - image: circleci/python:2.7
+        environment:
+          DJANGO=Django>=1.11,<2.0
+  py33dj18:
+    <<: *common
+    docker:
+      - image: circleci/python:3.3
+        environment:
+          DJANGO=Django>=1.8,<1.9
+  py34dj18:
+    <<: *common
+    docker:
+      - image: circleci/python:3.4
+        environment:
+          DJANGO=Django>=1.8,<1.9
+  py34dj19:
+    <<: *common
+    docker:
+      - image: circleci/python:3.4
+        environment:
+          DJANGO=Django>=1.9,<1.10
+  py34dj110:
+    <<: *common
+    docker:
+      - image: circleci/python:3.4
+        environment:
+          DJANGO=Django>=1.10,<1.11
+  py34dj111:
+    <<: *common
+    docker:
+      - image: circleci/python:3.4
+        environment:
+          DJANGO=Django>=1.11,<2.0
+  py34djmaster:
+    <<: *common
+    docker:
+      - image: circleci/python:3.4
+        environment:
+          DJANGO=https://github.com/django/django/tarball/master
+  py35dj18:
+    <<: *common
+    docker:
+      - image: circleci/python:3.5
+        environment:
+          DJANGO=Django>=1.8,<1.9
+  py35dj19:
+    <<: *common
+    docker:
+      - image: circleci/python:3.5
+        environment:
+          DJANGO=Django>=1.9,<1.10
+  py35dj110:
+    <<: *common
+    docker:
+      - image: circleci/python:3.5
+        environment:
+          DJANGO=Django>=1.10,<1.11
+  py35dj111:
+    <<: *common
+    docker:
+      - image: circleci/python:3.5
+        environment:
+          DJANGO=Django>=1.11,<2.0
+  py35djmaster:
+    <<: *common
+    docker:
+      - image: circleci/python:3.5
+        environment:
+          DJANGO=https://github.com/django/django/tarball/master
+  py36dj111:
+    <<: *common
+    docker:
+      - image: circleci/python:3.6
+        environment:
+          DJANGO=Django>=1.11,<2.0
+  py36djmaster:
+    <<: *common
+    docker:
+      - image: circleci/python:3.6
+        environment:
+          DJANGO=https://github.com/django/django/tarball/master
+
+workflows:
+  version: 2
+  lint-and-test:
+    jobs:
+      - lint
+      - py27dj18:
+          requires:
+            - lint
```

Comment:
Should "lint" be a requirement really?  (assuming that it will not run the other builds then)

Reply:
what that tells circle CI is to run the linting first before starting the testing.

<img width="471" alt="screen shot 2017-09-29 at 10 59 23 am" src="https://user-images.githubusercontent.com/1192/31024652-61d10192-a505-11e7-85c6-23be2b80b316.png">

Reply:
And if linting fails?

Reply:
then it doesn't bother running the tests.  you fix your lints and push again.

Reply:
well, that seems to be a bad feedback loop to me.
But:yeah for keeping the project on track!

https://user-images.githubusercontent.com/1192/30399847-a9cd658e-989a-11e7-85e7-264e87b603e7.gif

Reply:
Why not fail fast?